### PR TITLE
Added missing filter functionality in distinct operator

### PR DESCRIPTION
**Pull request by drorasaf**
Fixed issue on python 3.4 testing - import mock

### CODE REVIEW

File: tests/test__collection_api.py [hunk 1]
```diff
@@ -86,17 +86,25 @@ def test__distinct_nested_field(self):
...
         cursor = self.db.collection.find()
         self.assertEqual(cursor.distinct('f1.f2'), ['v'])
+        self.assertEqual(self.db.collection.distinct('f1.f2'), ['v'])
 
     def test__distinct_array_field(self):
         self.db.collection.insert(
             [{'f1': ['v1', 'v2', 'v1']}, {'f1': ['v2', 'v3']}])
         cursor = self.db.collection.find()
         self.assertEqual(set(cursor.distinct('f1')), set(['v1', 'v2', 'v3']))
+        self.assertEqual(set(self.db.collection.distinct('f1')), set(['v1', 'v2', 'v3']))
```
Comment (srinivasreddy) [inline tests/test__collection_api.py]: you can remove this line,  line 95,96 looks similar to me.

File: tests/test__collection_api.py [hunk 2]
```diff
@@ -86,17 +86,25 @@ def test__distinct_nested_field(self):
...
 
     def test__distinct_document_field(self):
         self.db.collection.insert({'f1': {'f2': 'v2', 'f3': 'v3'}})
         cursor = self.db.collection.find()
         self.assertEqual(cursor.distinct('f1'), [{'f2': 'v2', 'f3': 'v3'}])
+        self.assertEqual(self.db.collection.distinct('f1'), [{'f2': 'v2', 'f3': 'v3'}])
```
Comment (srinivasreddy) [inline tests/test__collection_api.py]: same here

**Comment by srinivasreddy** [comment]
@drorasaf  could you please squash these commits?

**Comment by drorasaf** [comment]
squashed

**Comment by srinivasreddy** [comment]
Thanks

**Pull request merged**